sycl/test-e2e/lit.cfg.py

```diff
@@ -821,6 +826,70 @@ def remove_level_zero_suffix(devices):
 if config.test_mode != "build-only":
     config.sycl_build_targets = set()
 
+# A device filter such as level_zero:gpu can have multiple devices under it and
+# the order is not guaranteed. The aspects enabled are also restricted to what
+# is supported on all devices under the label. It is possible for level_zero:gpu
+# and level_zero:0 to select different devices on different machines with the
+# same hardware. It is not currently possible to pass the device architecture to
+# ONEAPI_DEVICE_SELECTOR. Instead, if "BACKEND:arch-DEVICE_ARCH" is provided to
+# "sycl_devices", giving the desired device architecture, select a device that
+# matches that architecture using the backend:device-num device selection
+# scheme.
+filtered_sycl_devices = []
+for sycl_device in remove_level_zero_suffix(config.sycl_devices):
+
+    backend, device_arch  = sycl_device.split(":", 1)
+
+    if ("arch-" in device_arch):
```

Comment:
Invert the condition and do early continue. Alternatively, outline arch-processing and use lists comprehension like
```
tmp = [process_arch(sycl_dev) if sycl_dev.contains(":arch-") else sycl_dev
       for sycl_dev in remove_level_zero_suffix(config.sycl_devices)]
```